cmd/bosun/expr/expr_test.go

```diff
@@ -224,3 +224,31 @@ func TestQueryExpr(t *testing.T) {
 		}
 	}
 }
+
+func TestScalarPromotion(t *testing.T) {
+	tests := map[string]map[string]Value{
```

Comment:
this is a really confusing data structure to only hold one test. Either add more or inline the fields (or make an anonymous struct for test cases).